### PR TITLE
Add support for deployment protection rules

### PR DESCRIPTION
**Pull request by abhishekbiyala**
Fixes https://github.com/google/go-github/issues/3022

### CODE REVIEW

File: github/repos_deployment_protection_rules.go
Comment (gmlewis) [inline github/repos_deployment_protection_rules.go:1]:
```suggestion
// Copyright 2024 The go-github AUTHORS. All rights reserved.
```

File: github/repos_deployment_protection_rules.go
Comment (gmlewis) [inline github/repos_deployment_protection_rules.go:13]: Please add a descriptive godoc-style comment for this exported struct.

File: github/repos_deployment_protection_rules.go
Comment (gmlewis) [inline github/repos_deployment_protection_rules.go:28]:
```suggestion
// ListDeploymentProtectionRuleResponse represents the response that comes back when you list deployment protection rules.
```

File: github/repos_deployment_protection_rules.go
Comment (gmlewis) [inline github/repos_deployment_protection_rules.go:34]:
```suggestion
// ListCustomDeploymentRuleIntegrationsResponse represents the slightly different response that comes back when you list custom deployment rule integrations.
```

File: github/repos_deployment_protection_rules.go
Comment (gmlewis) [inline github/repos_deployment_protection_rules.go:37]: How about "Apps" => "AvailableIntegrations"?

File: github/repos_deployment_protection_rules.go
Comment (gmlewis) [inline github/repos_deployment_protection_rules.go:45]:
```suggestion
// GetAllDeploymentProtectionRules gets all the deployment protection rules for an environment.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @abhishekbiyala !
Just a few minor tweaks, please, then we will be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Review by gmlewis — APPROVED** [review]
Thank you, @abhishekbiyala !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by fchimpan** [comment]
Hi, @abhishekbiyala 
Thanks your implementation. LGTM!

**Comment by gmlewis** [comment]
Thank you, @fchimpan !
Merging.

**Pull request merged**
